api/src/controllers/userBooksController.js

```diff
@@ -6,373 +6,359 @@ import { buildUserBookDto } from "../services/userBookService.js";
+                    )
+                    .first();
+
+                return res.status(200).json({
+                    message: "Book status updated successfully.",
+                    book: buildUserBookDto(updatedBook),
+                });
+            }
+
+            await trx("UserBooks").insert({
+                user_id: req.user.userId,
+                book_id: book.book_id,
+                status,
+                start_date,
+                end_date,
+            });
+
+            return await trx("Books")
+                .join("UserBooks", "Books.book_id", "=", "UserBooks.book_id")
+                .where({
+                    "UserBooks.user_id": req.user.userId,
+                    "Books.book_id": book.book_id,
+                })
+                .select("Books.*", "UserBooks.status", "UserBooks.start_date", "UserBooks.end_date")
+                .first();
         });
-        return;
-      }
-
-      if (existingUserBook && existingUserBook.status !== status) {
-        await trx("UserBooks")
-          .where({
-            user_id: req.user.userId,
-            book_id: book.book_id,
-          })
-          .update({
-            status,
-            start_date,
-            end_date,
-          });
-
-        const updatedBook = await trx("Books")
-          .join("UserBooks", "Books.book_id", "=", "UserBooks.book_id")
-          .where({
-            "UserBooks.user_id": req.user.userId,
-            "Books.book_id": book.book_id,
-          })
-          .select(
-            "Books.*",
-            "UserBooks.status",
-            "UserBooks.start_date",
-            "UserBooks.end_date"
-          )
-          .first();
 
-        return res.status(200).json({
-          message: "Book status updated successfully.",
-          book: buildUserBookDto(updatedBook),
+        if (!result) {
+            return res.status(500).json({ error: "Failed to add book to the user's library." });
+        }
+
+        res.status(201).json({
+            message: "Book added to user's library successfully",
+            book: buildUserBookDto(result),
         });
-      }
-
-      await trx("UserBooks").insert({
-        user_id: req.user.userId,
-        book_id: book.book_id,
-        status,
-        start_date,
-        end_date,
-      });
-
-      return await trx("Books")
-        .join("UserBooks", "Books.book_id", "=", "UserBooks.book_id")
-        .where({
-          "UserBooks.user_id": req.user.userId,
-          "Books.book_id": book.book_id,
-        })
-        .select(
-          "Books.*",
-          "UserBooks.status",
-          "UserBooks.start_date",
-          "UserBooks.end_date"
-        )
-        .first();
-    });
-
-    if (!result) {
-      return res
-        .status(500)
-        .json({ error: "Failed to add book to the user's library." });
+        return;
+    } catch (error) {
+        console.error("Error adding book to user's library:", error);
```

Comment:
hm, why did you decide to skip returning 500 here?

Reply:
![1](https://github.com/user-attachments/assets/da42987e-9d9e-44c4-a233-940419d6b4a5)
it was like this already